Multivariable polynomial functors

fredrik-bakke opened the pull request.

VojtechStep commented on src/trees/multivariable-polynomial-functors.lagda.md:
```diff
@@ -0,0 +1,372 @@
+            coherence-triangle-maps (pr2 (x j) i) (y i) (tr (B i {j}) p)}
+          ( is-torsorial-Id (pr1 (x j)))
+          ( pr1 (x j) , refl)
+          (is-torsorial-binary-htpy (pr2 (x j))))
```
```suggestion
          ( is-torsorial-binary-htpy (pr2 (x j))))
```

VojtechStep commented on src/trees/multivariable-polynomial-functors.lagda.md:
```diff
@@ -0,0 +1,372 @@
+  Eq-eq-type-polynomial-functor x .x refl =
+    refl-Eq-type-polynomial-functor x
+
+  is-torsorial-Eq-type-polynomial-functor :
```
Please make this abstract

VojtechStep commented on src/trees/multivariable-polynomial-functors.lagda.md:
```diff
@@ -0,0 +1,372 @@
+          ( pr1 (x j) , refl)
+          (is-torsorial-binary-htpy (pr2 (x j))))
+
+  is-equiv-Eq-eq-type-polynomial-functor :
```
Please make this abstract

VojtechStep commented on src/trees/multivariable-polynomial-functors.lagda.md:
```diff
@@ -0,0 +1,372 @@
+    {l8 l9 : Level} {X : I → UU l8} {Y : I → UU l9}
+    (f : (i : I) → X i → Y i) (k : K) →
+    ( map-polynomial-functor comp-polynomial-functor f k) ~
+    ( map-inv-compute-type-comp-polynomial-functor Y k) ∘
+    ( map-polynomial-functor 𝑄 (map-polynomial-functor 𝑃 f) k) ∘
+    ( map-compute-type-comp-polynomial-functor X k)
```
```suggestion
    ( map-inv-compute-type-comp-polynomial-functor Y k ∘
      map-polynomial-functor 𝑄 (map-polynomial-functor 𝑃 f) k ∘
      map-compute-type-comp-polynomial-functor X k)
```
I'd write it like this, since the purpose of the coding style with infix operators is to see at least the application tree

VojtechStep commented on src/trees/multivariable-polynomial-functors.lagda.md:
```diff
@@ -0,0 +1,372 @@
+    {l8 l9 : Level} {X : I → UU l8} {Y : I → UU l9}
+    (f : (i : I) → X i → Y i) (k : K) →
+    ( map-polynomial-functor 𝑄 (map-polynomial-functor 𝑃 f) k) ~
+    ( map-compute-type-comp-polynomial-functor Y k) ∘
+    ( map-polynomial-functor comp-polynomial-functor f k) ∘
+    ( map-inv-compute-type-comp-polynomial-functor X k)
```
```suggestion
    ( map-compute-type-comp-polynomial-functor Y k ∘
      map-polynomial-functor comp-polynomial-functor f k ∘
      map-inv-compute-type-comp-polynomial-functor X k)
```

VojtechStep reviewed (commented):
Very nice! I unfortunately know very little about polynomial functors/containers, so I mostly just checked the PR on a surface level. Could you cite a standard reference for the development?

fredrik-bakke commented on src/foundation/structure.lagda.md:
```diff
@@ -64,3 +64,7 @@ has-structure-equiv' :
+
+## See also
+
+- [Species of types](species.species-of-types.md)
```
I added this link because I almost ended up reimplementing something that was already done for _species of types_, and not _structure_. Although, they are, ignoring attitude, the same thing.

VojtechStep replied:
I feel like something existing for species but not structure is subpar factoring, since "The Thing" ought to be more fundamental than "The Thing viewed a Certain Way". But it's not so important to hold this PR up, so I'm resolving this

fredrik-bakke commented:
> Very nice! I unfortunately know very little about polynomial functors/containers, so I mostly just checked the PR on a surface level. Could you cite a standard reference for the development?

Thanks for the review! I've addressed your comments now
